### DISCUSSION TITLE
Independent deployment of Sunbird Lern

### DISCUSSION THREAD

**Discussion by sagarihanda** [Sunbird-Lern/Community#33] [Query]
### Project Context
UP State Medical Faculty (SMF) is building a platform to manage human resources for allied healthcare professionals (nurses, paramedical staff, lab technicians etc.) in the state. The Government is looking to build the platform on Sunbird building blocks to adhere to ABDM's (https://abdm.gov.in/abdm) principles. One of the key modules being built is to enable competency based assessments of allied professionals, and to enable mid-career learning courses via an LMS. Six Sunbird blocks (RC, Obsrv, Knowlg, inQuiry, CoKreat and Ed) are being leveraged to develop the platform with the user data being housed in Sunbird RC.

### Requirements
A Lern powered LMS is being conceptualized for students in allied healthcare courses and working professionals to enable them to upskill on identified competency areas. The following workflows are being envisioned on the LMS platform:

- A **_content provider_** can upload content pieces (video, PDF, assessment) and organize into course modules
- A **_content provider_** can tag competencies to course modules
- A **_content provider_** can save a course and send it to **_admin_** for review
- **_Admin_** can review course content and relevance of competency tags
- **_Admin_** can approve and publish courses for discovery by students
- **_Admin_** can reject courses / send back for updations to **_content provider_**
- **_Student_** can register on the platform and enter their position or job role
- **_Student_** can view activities, competencies required for their position or job role
- **_Student_** can attempt self assessments to identify weak competencies
- **_Student_** can view course recommendations as per identified competencies
- **_Student_** can save courses to attempt at a later time
- **_Student_** can progress through content under a course (videos, PDFs, assessments)
- **_Student_** can attempt an assessment post course completion
- **_Student_** can receive and save a badge / certificate on successful completion of course

### Question
Are there dependencies that will arise on deploying Sunbird Lern independently for these workflows? 

@ChakshuGautam @rhwarrier @rahul101001000 @reshmi-nair @pkranga @mahesh-maney @maheshkumargangula @vrayulu

**Comment by reshmi-nair** [comment]
@sagarihanda Sunbird Lern has dependencies on Sunbird Knowlg and Sunbird Obsrv. Also for certificate generation Sunbird RC is being used. 
Batch service in Sunbird Lern internally uses UserOrg service for fetching user details, user validations and sending notifications. This will have to be decoupled if user data is going to be stored in Sunbird RC.
cc: @rhwarrier @maheshkumargangula @vrayulu
